docs/source/java/cdata.rst

```diff
@@ -0,0 +1,313 @@
+
+        // Release callback
+        void (*release)(struct ArrowArray*);
+        // Opaque producer-specific data
+        void* private_data;
+    };
+
+    static void release_int32_type(struct ArrowSchema* schema) {
+        // Mark released
+        schema->release = NULL;
+    }
+
+    void export_int32_type(struct ArrowSchema* schema) {
+        *schema = (struct ArrowSchema) {
+                // Type description
+                .format = "l",
+                .name = "",
+                .metadata = NULL,
+                .flags = 0,
+                .n_children = 0,
+                .children = NULL,
+                .dictionary = NULL,
+                // Bookkeeping
+                .release = &release_int32_type
+        };
+        std::cout << "C Data - Schema Pointer = " << schema << std::endl;
+    }
+
+    static void release_int32_array(struct ArrowArray* array) {
+        assert(array->n_buffers == 2);
+        // Free the buffers and the buffers array
+        free((void *) array->buffers[1]);
+        free(array->buffers);
+        // Mark released
+        array->release = NULL;
+    }
+
+    void export_int32_array(struct ArrowArray* array) {
+        arrow::Int64Builder builder;
+        builder.Append(1);
+        builder.Append(2);
+        builder.Append(3);
+        builder.AppendNull();
+        builder.Append(5);
+        builder.Append(6);
+        builder.Append(7);
+        builder.Append(8);
+
+        auto maybe_array = builder.Finish();
+        std::shared_ptr<arrow::Array> array_arrow = *maybe_array;
+        auto int64_array = std::static_pointer_cast<arrow::Int64Array>(array_arrow);
+        const int64_t* data = int64_array->raw_values();
+        std::cout << "Data To Exchange Pointer = " << data << std::endl;
+        for (int j = 0; j < int64_array->length(); j++){
+            std::cout << "Data To Exchange Value[" << j << "] = " << data[j] << std::endl;
+        }
+
+        *array = (struct ArrowArray) {
+                // Data description
+                .length = int64_array->length(),
+                .offset = 0,
+                .null_count = 0,
+                .n_buffers = 2,
+                .n_children = 0,
+                .children = NULL,
+                .dictionary = NULL,
+                // Bookkeeping
+                .release = &release_int32_array
+        };
+
+        // Allocate list of buffers
+        array->buffers = (const void**) malloc(sizeof(void*) * array->n_buffers);
+        assert(array->buffers != NULL);
+        array->buffers[0] = NULL;  // no nulls, null bitmap can be omitted
+        array->buffers[1] = data;
+
+        std::cout << "C Data - Array Pointer = " << array << std::endl;
+        std::cout << "C Data - Array Data Pointer Buffer array->buffers[1] = " << array->buffers[1] << std::endl;
+    }
+
+**Java Side**
+
+Define Java code CDataInterfaceLibraryConfig.java that consume by JNI C++ functions exported through
+C Data Interface:
```

Comment:
"For this example, we will use javacpp (link to project) to call our main C++ function from Java, without writing JNI bindings ourselves."